packages/next/src/server/app-render/app-render.tsx

```diff
@@ -317,955 +530,44 @@ const wrappedRender = async (
-              rootLayoutIncluded: rootLayoutIncludedAtThisLevelOrAbove,
-              asNotFound,
-              metadataOutlet,
-            })
-
-            return path
-              .map((item) => {
-                // we don't need to send over default routes in the flight data
-                // because they are always ignored by the client, unless it's a refetch
-                if (
-                  item[0] === '__DEFAULT__' &&
-                  flightRouterState &&
-                  !!flightRouterState[1][parallelRouteKey][0] &&
-                  flightRouterState[1][parallelRouteKey][3] !== 'refetch'
-                ) {
-                  return null
-                }
-                return [actualSegment, parallelRouteKey, ...item]
-              })
-              .filter(Boolean) as FlightDataPath[]
-          })
-        )
-      ).flat()
-
-      return paths
-    }
-
-    // Flight data that is going to be passed to the browser.
-    // Currently a single item array but in the future multiple patches might be combined in a single request.
-
-    let flightData: FlightData | null = null
-    if (!options?.skipFlight) {
-      const [MetadataTree, MetadataOutlet] = createMetadataComponents({
-        tree: loaderTree,
-        pathname: urlPathname,
-        searchParams: providedSearchParams,
-        getDynamicParamFromSegment,
-        appUsingSizeAdjust,
-      })
-      flightData = (
-        await walkTreeWithFlightRouterState({
-          createSegmentPath: (child) => child,
-          loaderTreeToFilter: loaderTree,
-          parentParams: {},
-          flightRouterState: providedFlightRouterState,
-          isFirst: true,
-          // For flight, render metadata inside leaf page
-          rscPayloadHead: (
-            // Adding requestId as react key to make metadata remount for each render
-            <MetadataTree key={requestId} />
-          ),
-          injectedCSS: new Set(),
-          injectedFontPreloadTags: new Set(),
-          rootLayoutIncluded: false,
-          asNotFound: isNotFoundPath || options?.asNotFound,
-          metadataOutlet: <MetadataOutlet />,
-        })
-      ).map((path) => path.slice(1)) // remove the '' (root) segment
-    }
-
-    const buildIdFlightDataPair = [buildId, flightData]
-
-    // For app dir, use the bundled version of Flight server renderer (renderToReadableStream)
-    // which contains the subset React.
-    const flightReadableStream = renderToReadableStream(
-      options
-        ? [options.actionResult, buildIdFlightDataPair]
-        : buildIdFlightDataPair,
-      clientReferenceManifest.clientModules,
-      {
-        context: serverContexts,
-        onError: flightDataRendererErrorHandler,
-      }
-    ).pipeThrough(createBufferedTransformStream())
+  const getDynamicParamFromSegment = makeGetDynamicParamFromSegment(
+    params,
+    providedFlightRouterState
+  )
 
-    return new FlightRenderResult(flightReadableStream)
```

Comment:
For reviewers: this is the derived context being passed around in the sub-functions